### ISSUE TITLE
Question: Linux ARM - missing DeviceInfoDetails (serial, description, ...)

### ISSUE THREAD

**Issue by lovetheguitar** [snmishra/ftd2xx#42]
Hi all,

I have a question regarding the ftd2xx (driver) on a RasPi 4B (2GB) running Ubuntu Server 20.04.3 LTS.
I installed the driver `libftd2xx-arm-v8-1.4.24.tgz`, following the instructions inside the driver's ReadMe.

Problem is, that no matter how I try I always get an almost empty DeviceInfoDetail from my USB device.
I already tried different version of this library here and connecting the device to all ports (USB2 and 3) of the RasPi.

The device details always look like this: 
```python
{'index': 0, 'flags': 1, 'type': 3, 'id': 0, 'location': 0, 'serial': b'', 'description': b'', 'handle': <ftd2xx._ftd2xx_linux.LP_c_ulong object at 0xffffbab95d40>}
```
On a windows PC I correctly get the description etc. 

I'm very grateful for any hints.

Thank you.

PS: Found the same question also here (no answer) https://forums.raspberrypi.com/viewtopic.php?t=85436.

**Comment by snmishra** [comment]
Hello @lovetheguitar Thanks for trying on a new platform! I don't have access to the platform, so if you are willing to work with me, we might be able to solve this.

The relevant line that loads libftd2xx is `_libraries["libftd2xx.so"] = CDLL("libftd2xx.so")` in [_ftd2xx_linux.py](https://github.com/snmishra/ftd2xx/blob/625d727a932c5e42f9cd2923df6faadafb6f714f/ftd2xx/_ftd2xx_linux.py#L7)

So the driver name needs to match this be available where ctypes can find it. That is typically somewhere on your `$LD_LIBRARY_PATH`.

In addition your user must have ability to access the device file (e.g. /dev/usb0).

**Comment by krister-ts** [comment]
@snmishra I have tested with Pi OS (32-bit) and arm v7 driver. Looks fine. I'm still waiting for my other Pi to come back which has ubuntu 64-bit and arm v8 driver (same as OP). I will let you know if there is an issue when I get it back.

**Comment by lovetheguitar** [comment]
Dear @snmishra,

sorry for the late reply and thanks a lot for your willingness to help.

My problem was, that I did not understand that I had to _permanently_ block the `usbserial` and `ftdi_sio` drivers in /etc/modprobe.d/blacklist.conf.


> In addition your user must have ability to access the device file (e.g. /dev/usb0).

That is another good hint, a quick "sanity check",  by running my script with `sudo`, showed everything working as expected.

Thanks a lot again. ❤️
